File: ios/NativeModuleDemo/Calculator.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE (Calculator, NSObject)
RCT_EXTERN_METHOD(add
                  : (nonnull NSNumber *)a b
                  : (nonnull NSNumber *)b callback
                  : (RCTResponseSenderBlock)callback)
@end

File: ios/NativeModuleDemo/Calculator.swift
import Foundation
import React

@objc(CalculatorModule)
class CalculatorModule: NSObject {

    @objc
    func add(_ a: NSNumber, b: NSNumber, callback: @escaping RCTResponseSenderBlock) {
        let sum = a.doubleValue + b.doubleValue
        callback([NSNull(), sum])  // Sending the result back to JavaScript
    }

    static func requiresMainQueueSetup() -> Bool {
        return false
    }
}
